plugins/plugin-docker/che-plugin-openshift-client/src/main/java/org/eclipse/che/plugin/openshift/client/OpenShiftConnector.java

```diff
@@ -967,6 +932,47 @@ private ContainerInfo createContainerInfo(Service svc,
         return info;
     }
 
+
+    private void cleanUpWorkspaceResources(String deploymentName) throws IOException {
+
+        Deployment deployment = getDeploymentByName(deploymentName);
+        Service service = getCheServiceBySelector(OPENSHIFT_DEPLOYMENT_LABEL, deploymentName);
+
+        if (service != null) {
+            LOG.info("Removing OpenShift Service {}", service.getMetadata().getName());
+            openShiftClient.resource(service).delete();
+        }
+
+        if (deployment != null) {
+            LOG.info("Removing OpenShift Deployment {}", deployment.getMetadata().getName());
+            openShiftClient.resource(deployment).delete();
+        }
+
+        // Wait for all pods to terminate before returning.
+        List<Pod> pods = openShiftClient.pods()
+                                        .inNamespace(openShiftCheProjectName)
+                                        .withLabel(OPENSHIFT_DEPLOYMENT_LABEL, deploymentName)
+                                        .list()
+                                        .getItems();
+        try {
+            boolean podsTerminated = false;
+            while (!podsTerminated) {
```

Comment:
Fixed. Although, if a pod never terminates, the corresponding workspace will be blocked from launching, breaking a fair bit of Che.